### ISSUE TITLE
Is reporting radio buttons with required attribute as errors not counter to HTML spec?

### ISSUE THREAD

**Issue by alanfluff** [ThePacielloGroup/ARC-Toolkit#32]
Auditing a page with a set of radios using `required`, i.e.: `<input type="radio" ... required>` results in ARC Toolkit reporting errors (e.g. below), yet [the HTML spec notes](https://html.spec.whatwg.org/multipage/input.html#the-required-attribute:~:text=To%20avoid%20confusion%20as%20to%20whether%20a%20radio%20button%20group%20is%20required%20or%20not%2C%20authors%20are%20encouraged%20to%20specify%20the%20attribute%20on%20all%20the%20radio%20buttons%20in%20a%20group):

> To avoid confusion as to whether a [radio button group](https://html.spec.whatwg.org/multipage/input.html#radio-button-group) is required or not, authors are encouraged to specify the attribute on all the radio buttons in a group.

## ARC Toolkit error example
> 2  Error(s) aria-required or required attribute on radio button
> Description: The aria-required or required attribute is used on a radio button.
> WCAG 2.1 Success Criterion:A 3.3.2 Labels or Instructions 

Unless I have interpreted the spec' incorrectly, should ARC Toolkit not allow the required` attribute on radios without flagging errors?

- ARC Toolkit Chrome Extension v5.5.3
- Chrome 111.0.5563.147 on Windows 10

**Comment by stevefaulkner** [comment]
@alanfluff  You are correct, `aria-required` is not allowed on [`role=radio`](https://www.w3.org/TR/wai-aria-1.2/#radio), but `required` is allowed on `input type="radio"`

@ferllings  @poorgeek this is a bug

**Comment by ferllings** [comment]
This is 'old' rule. I'm going to deprecate it and create a new rule to handle `aria-required` not allowed on `role=radio`

**Comment by alanfluff** [comment]
> This is 'old' rule. I'm going to deprecate it and create a new rule to handle `aria-required` not allowed on `role=radio`

I assume your change will include no longer marking an error when the the HTML5 `required` attribute is present on a `radio` `input`.

Also, if your UI allows for it, it would be useful to note [the recommendation about `required` being on **all** `radios` in the same group](https://html.spec.whatwg.org/multipage/input.html#the-required-attribute:~:text=To%20avoid%20confusion%20as%20to%20whether%20a%20radio%20button%20group%20is%20required%20or%20not%2C%20authors%20are%20encouraged%20to%20specify%20the%20attribute%20on%20all%20the%20radio%20buttons%20in%20a%20group).
